makeflow/src/makeflow.c

```diff
@@ -177,6 +178,74 @@ static struct list *shared_fs_list = NULL;
 
 static int did_find_archived_job = 0;
 
+static struct list *jx_argv = NULL;
+static struct jx *jx_optarg = NULL;
+
+static void push_jx_argv(struct jx *j) {
+	assert(j);
+	if (!jx_argv)
+			jx_argv = list_create();
+	list_push_head(jx_argv, j);
+}
+
+int option_from_name(const struct option *opt, const char *name, int *indexptr) {
+	assert(opt);
+	assert(name);
+	for (int i = 0; opt[i].name; ++i) {
+		if (!strcmp(name, opt[i].name)) {
+			if (indexptr)
+				*indexptr = i;
+			return opt[i].val;
+		}
+	}
+	return 0;
+}
+
+char *optarg_from_jx(struct jx *j) {
+	if (!j) return NULL;
+	switch (j->type) {
+	case JX_BOOLEAN:
+		return j->u.boolean_value ? xxstrdup("true") : xxstrdup("false");
+	case JX_INTEGER:
+		return string_format("%" PRIi64, j->u.integer_value);
+	case JX_DOUBLE:
+		return string_format("%g", j->u.double_value);
+	case JX_STRING:
+		return xxstrdup(j->u.string_value);
+	default:
+		return NULL;
+	}
+}
+
+static int makeflow_getopt(int argc, char *const argv[], const char *optstring, const struct option *longopts, int *longindex) {
```

Comment:
All these are option related functions seem generic. I would suggest to move them to a jx_options.c in dttools. We can then use them somewhere else (e.g. to not worry about quoting when wrapping the resource_monitor).

Reply:
Agreed!